com.unity.ml-agents/Tests/Editor/MLAgentsEditModeTest.cs

```diff
@@ -559,5 +575,31 @@ public void TestMaxStepsReset()
                 Assert.AreEqual(expectedCollectObsCallsSinceReset, agent1.collectObservationsCallsSinceLastReset);
             }
         }
+
+        [Test]
+        public void TestHeuristicPolicyStepsSensors()
+        {
+            // Make sure that Agents with HeuristicPolicies step their sensors each Academy step.
+            var agentGo1 = new GameObject("TestAgent");
+            agentGo1.AddComponent<TestAgent>();
+            var agent1 = agentGo1.GetComponent<TestAgent>();
+            var aca = Academy.Instance;
+
+            var decisionRequester = agent1.gameObject.AddComponent<DecisionRequester>();
+            decisionRequester.DecisionPeriod = 1;
+            decisionRequester.Awake();
+
+            agent1.LazyInitialize();
+            Assert.AreEqual(agent1.GetPolicy().GetType(), typeof(HeuristicPolicy));
```

Comment:
It turns out that most of the other Agents in this test are using Heuristic policies by default. I added checks specifically for this test to use that we get a HeuristicPolicy and that the heuristic is stepped.